### PR TITLE
420 document parallel iterator methods

### PR DESCRIPTION
**Pull request by oddg**
Contribution for closing https://github.com/rayon-rs/rayon/issues/420.

There is one method left in the `ParallelIterator` trait that has no example: `opt_len`.
Also the example for `fold_with` might not be necessary.

### CODE REVIEW

File: src/iter/mod.rs [hunk 1]
```diff
@@ -266,6 +266,23 @@ pub trait ParallelIterator: Sized + Send {
...
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::mpsc::channel;
+    /// use rayon::prelude::*;
+    ///
+    /// let (sender, receiver) = channel();
+    ///
+    /// let mut par_iter = (0..5).into_par_iter().map_with(sender, |s, x| { s.send(x).unwrap(); x });
```
Comment (nikomatsakis) [inline src/iter/mod.rs]: Nit: can we format this a bit more clearly? Maybe run the example through rustfmt (on play) and paste it back? I feel like it's a bit hard to follow what's happening. (Also, some comments explaining what's going on wouldn't hurt.)

File: src/iter/mod.rs [hunk 2]
```diff
@@ -484,6 +501,18 @@ pub trait ParallelIterator: Sized + Send {
     /// efficient. If possible, it is better to call `reduce()`, which
     /// requires an identity element.
     ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    /// let sums = [(0, 1), (5, 6), (16, 2), (8, 9)]
+    ///            .par_iter()        // iterating over &(i32, i32)
+    ///            .cloned()          // iterating over (i32, i32)
```
Comment (nikomatsakis) [inline src/iter/mod.rs]: formatting like this is great =)

**Comment by oddg** [comment]
@nikomatsakis Thanks for your review.
I hope the example is clearer now.

**Comment by cuviper** [comment]
bors r=nikomatsakis
